seed/analysis_pipelines/bsyncr.py

```diff
@@ -0,0 +1,254 @@
+            _prepare_all_properties.s(analysis_id, progress_data.key),
+            _finish_preparation.si(analysis_id, progress_data.key)
+        ).apply_async()
+
+        return progress_data.key
+
+
+@shared_task
+def _prepare_all_properties(analysis_property_view_ids, analysis_id, progress_data_key):
+    """A Celery task which attempts to make BuildingSync files for all AnalysisPropertyViews.
+
+    :param analysis_property_view_ids: list[int]
+    :param analysis_id: int
+    :param progress_data_key: str
+    :returns: void
+    """
+    progress_data = ProgressData.from_key(progress_data_key)
+    progress_data.step('Creating files for analysis')
+
+    analysis = Analysis.objects.get(id=analysis_id)
+    analysis_property_views = AnalysisPropertyView.objects.filter(id__in=analysis_property_view_ids)
+    input_file_paths = []
+    for analysis_property_view in analysis_property_views:
+        try:
+            meters = (
+                Meter.objects
+                .annotate(readings_count=Count('meter_readings'))
+                .filter(
+                    property=analysis_property_view.property,
+                    type__in=[Meter.ELECTRICITY_GRID, Meter.ELECTRICITY_SOLAR, Meter.ELECTRICITY_WIND],
+                    readings_count__gte=12,
+                )
+            )
+            if meters.count() == 0:
+                AnalysisMessage.objects.create(
+                    analysis=analysis,
+                    analysis_property_view=analysis_property_view,
+                    type=AnalysisMessage.DEFAULT,
+                    user_message='Property has no linked electricity meters with 12 or more readings'
+                )
+                continue
+
+            # arbitrarily choosing the first meter for now
+            meter = meters[0]
+
+            bsync_doc, errors = _build_bsyncr_input(analysis_property_view, meter)
+            if errors:
+                for error in errors:
+                    AnalysisMessage.objects.create(
+                        analysis=analysis,
+                        analysis_property_view=analysis_property_view,
+                        type=AnalysisMessage.DEFAULT,
+                        user_message=error
+                    )
+                continue
+
+            analysis_input_file = AnalysisInputFile(
+                content_type=AnalysisInputFile.BUILDINGSYNC,
+                analysis=analysis
+            )
+            analysis_input_file.file.save(f'{analysis_property_view.id}.xml', ContentFile(bsync_doc))
+            analysis_input_file.clean()
+            analysis_input_file.save()
+            input_file_paths.append(analysis_input_file.file.path)
+        except Exception as e:
+            AnalysisMessage.objects.create(
+                analysis=analysis,
+                analysis_property_view=analysis_property_view,
+                type=AnalysisMessage.DEFAULT,
+                user_message='Unexpected error',
```

Comment:
Not sure what a better user message for this might be. Looking for feedback

Reply:
I think this is fine. We'll let the frontend handle any potential displaying of the debug message.